org.adempiere.base/src/org/adempiere/base/AbstractModelFactory.java

```diff
@@ -53,6 +53,8 @@ public static PO getPO(Class<?> clazz, String tableName, int Record_ID, String t
 		MTable table = MTable.get(Env.getCtx(), tableName);
 		if (table != null && table.isUUIDKeyTable())
 		{
+			if (Record_ID == 0 && ! MTable.isZeroIDTable(tableName)) // this is the convention to create a new record
```

Comment:
why is "! MTable.isZeroIDTable(tableName)" test needed here if table.isUUIDKeyTable() == true ?